dashboard/http_server_head.py

```diff
@@ -119,19 +119,25 @@ async def metrics_middleware(self, request, handler):
         start_time = time.monotonic()
         try:
             response = await handler(request)
-            status_tag = f"{floor(response.status / 100)}xx"
+            try:
+                status_tag = f"{floor(response.status / 100)}xx"
+            except Exception:
+                status_tag = "unknown"
```

Comment:
If response is None, we should probably handle this differently. I feel like it's better raising an exception rather than returning None as well..

Reply:
I'm okay with this (at least to not cherry-pick this for 2.1) but my concern is that we may have previously relied on this silent fail behavior and there is functionality that now breaks...

Reply:
I feel like this may be okay since this was previously caught by raise Exception block and explicitly failed the request anyway. My suggestion is just to add 

```python3
            response = await handler(request)
            assert response is not None, ("some error")
            status_tag = f"{floor(response.status / 100)}xx"
```
Wdyt? I think this makes sense because handler should never return None theoretically